### DISCUSSION TITLE
multi hydra clients with different domains

### DISCUSSION THREAD

**Discussion by simonwjy** [ory/hydra#2733] [Q&A] [answered]
Hi Experts,

our project has a hydra service and initially we have one web application A and BE application with some domain. Later we try to create one more web application B with different domain and also another hydra client under this hydra service. so for now we have one hydra service with two different hydra clients and the url related to these two hydra clients are in different domains. we also config the issuer under the domain A.

if i understand correctly we can only config the url for login/logout/issuer/... per hydra service?  When we are in web application B and try to trigger the hydra call the redirect url from is in domain A which we cause the cors issue.

So my questions is that is it recommend to create multi hydra clients with different domains in one hydra service? based on the above scenario can we add **allowed_origins** in hydra config file to solve this issue?

Looking forward to your reply and thank you.

Best Regards,
Simon

**Comment by vinckr** [comment, marked as answer]
`Is it recommend to create multi hydra clients with different domains in one hydra service?`
A good way to solve this is to have one Hydra AuthZ server and one client that serves both domains A and B. 

If you want true multi-tenancy (individual OAuth2 clients per domain, …) you need multiple 
deployments of Ory Hydra.

`If i understand correctly we can only config the url for login/logout/issuer/... per hydra service?`

Yes, that is correct, but you have full control over the frontend/login app, so you can redirect your login app to the same Hydra login URL for example. 

I hope that clears things up a bit, multiple domains are always a bit of a hassle. Let me know if anything still remains unclear!

**Reply by simonwjy** [reply]
thx @vinckr , your information is very helpful!!
